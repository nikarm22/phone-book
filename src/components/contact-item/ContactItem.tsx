import { Avatar, ListItem, ListItemAvatar, ListItemText } from '@material-ui/core';
import React from 'react';
import IContact from '../../constants/types/contact';

export interface IContactItem {
  contact: IContact;
}

export default function ContactItem(props: IContactItem) {
  const { contact } = props;

  return (
    <ListItem>
      <ListItemAvatar>
        <Avatar>{contact.name.charAt(0).toUpperCase() || '#'}</Avatar>
      </ListItemAvatar>
      <ListItemText
        primary={contact.name}
        secondary={contact.number}
      />
    </ListItem>
  );
}
