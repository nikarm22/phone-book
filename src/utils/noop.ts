// mock empty function
// istanbul ignore next
export default function noop() {}
