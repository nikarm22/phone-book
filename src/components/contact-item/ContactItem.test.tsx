import React from 'react';
import { render } from '@testing-library/react';
import ContactItem from './ContactItem';
import noop from '../../utils/noop';
import IContact from '../../constants/types/contact';

const mockContact: IContact = {
  id: '123',
  name: 'test',
  number: '123',
};
const mockContactEmptyName: IContact = {
  id: '123',
  name: '',
  number: '123',
};

describe('components/ContactItem', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(<ContactItem contact={mockContact} onClick={noop} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should put \'#\' if the name is empty', () => {
    const { getByText } = render(<ContactItem contact={mockContactEmptyName} onClick={noop} />);
    expect(getByText('#')).toBeInTheDocument();
  });
});
