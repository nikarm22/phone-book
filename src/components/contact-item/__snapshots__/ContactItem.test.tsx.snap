// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ContactItem should match the snapshot 1`] = `
<DocumentFragment>
  <li
    class="MuiListItem-root MuiListItem-gutters"
  >
    <div
      class="MuiListItemAvatar-root"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
      >
        T
      </div>
    </div>
    <div
      class="MuiListItemText-root MuiListItemText-multiline"
    >
      <span
        class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        test
      </span>
      <p
        class="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        123
      </p>
    </div>
  </li>
</DocumentFragment>
`;
